mito-ai/src/Extensions/AiChat/ChatHistoryManager.tsx

```diff
@@ -134,6 +134,72 @@ Your task: ${input}`};
         this.history.aiOptimizedChatHistory.push(aiOptimizedMessage);
     }
 
+    updateMessageAtIndex(index: number, newContent: string): void {
+        const variables = this.variableManager.variables
+        const activeCellCode = getActiveCellCode(this.notebookTracker)
+
+        const aiOptimizedMessage: OpenAI.Chat.ChatCompletionMessageParam = {
```

Comment:
Let's turn this prompt into a function so we don't have it here and in the `addChatInputMessage` function.